tests/e2e/test_gateway_commands.py

```diff
@@ -36,7 +35,58 @@ def test_deploying_contract_with_constructor(
     )
 
     assert "Deploy transaction was sent" in result
-    assert count_hex64(result) == 2
+    assert re.search(r"Transaction hash: 0x[0-9a-f]{64}", result), result
+
+    m = re.search(r"Contract address: (0x[0-9a-f]{64})", result)
+    assert m, result
+    contract_address = m[1]
+
+    result = protostar(
+        [
+            "--no-color",
+            "call",
+            "--gateway-url",
+            devnet_gateway_url,
+            "--chain-id",
+            str(StarknetChainId.TESTNET.value),
+            "--contract-address",
+            contract_address,
+            "--function",
+            "get_balance",
+        ],
+        ignore_stderr=True,
+    )
+
+    assert (
+        result
+        == """\
+Call successful.
+Response:
+{
+    "res": 66
+}
+"""
+    )
+
+    result = protostar(
+        [
+            "call",
+            "--gateway-url",
+            devnet_gateway_url,
+            "--chain-id",
+            str(StarknetChainId.TESTNET.value),
+            "--contract-address",
+            contract_address,
+            "--function",
+            "get_balance",
+            "--json",
+        ],
+        ignore_stderr=True,
+    )
+
+    assert result == """\
```

Comment:
I'd parse it and compare dictionaries, since some whitespaces might change and the test is less volatile that way

Reply:
My idea is to also test whitespace. I want to ensure that output is not starting with whitespace, and the object is compressed as much as possible.